app/interface/deps/providers.py

```
@@ -20,7 +19,7 @@ def usecase_provider_factory(
     유즈케이스 클래스에 단일 의존성을 주입하는 FastAPI provider를 생성합니다.
     """
 
-    def _provider(dep_obj: any = Depends(dependency)) -> usecase_cls:
+    def _provider(dep_obj: any = Depends(dependency)) -> Callable:
```

Ruff (0.14.7)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)